### PR TITLE
Initial demo script

### PR DESCRIPTION
**Pull request by bitwiseguy**
Closes https://github.com/statechannels/SCBridge-Wallet/issues/121

### CODE REVIEW

File: demo.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+## Demo script:
+
+- ERC-4337 compatible Smart Contract wallet
+- Unique feature: gasless off-chain txs with instant finality enabled by HTLC technology, as used in Bitcoin Lightning
+- Setup required:
+  - wallet owners must designate an Intermediary in their SC wallet
+  - wallet owners and Intermediary must sign-off on some amount of wallet funds designated for off-chain txs
```
Comment (geoknee) [inline demo.md:7]: We should emphasize this is off chain.

File: demo.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  - run in-browser client to interact with SC wallet and peers
+  - `Explain browser tab setup with Alice(left), Irene(middle), Bob(right)`
+- How it works:
+  - on-chain txs are 2/2 multi-sig (user + intermediary)
+  - `Show on-chain tx from Alice to Metamask account`
+  - user can remove intermediary at any point to claim full ownership of wallet/funds
+  - `Point to 'Eject' button on UI`
+  - off-chain payments are multisig between payor, intermediary, and payee
```
Comment (geoknee) [inline demo.md:15]: I'm not sure "multisig" does this full justice. How about say that "it is a pair of atomic payments: one from payer to intermediary and one from intermediary to payee"?

File: demo.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  - off-chain payments require users to be connected to the same intermediary (or have a path to each other through multiple intermediaries)
+  - `Show off-chain payment Alice to Bob`
+- Cross-chain payments
+  - works for off-chain payments only
+  - Alice on Network1 can pay Bob on Network2 if the following conditions are met:
+    - Alice designates some of her Network1 funds to Intermediary
+    - Bob designates some of his Network2 funds Intermediary
```
Comment (geoknee) [inline demo.md:22]: "Alice has Network1 funds she can sign over to Irene, and Irene has Network2 funds she can sign over to Bob."

**Review by geoknee — APPROVED** [review]
I think this is a good start!

File: demo.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+## Demo script:
+
+- ERC-4337 compatible Smart Contract wallet
+- Unique feature: gasless off-chain txs with instant finality enabled by HTLC technology, as used in Bitcoin Lightning
```
Comment (geoknee) [inline demo.md:4]: As mentioned in standup, our contribution is a novel way to hybridize thees two without needing any on chain transactions to move funds in or out of a layer 2 system.

File: demo.md (hunk 3 above)
Comment (bitwiseguy) [reply demo.md:22]: I was trying to list requirements for off-chains payments to work. Can you rephrase your statement so that it indicates what MUST be done instead of what COULD be done?

Comment (bitwiseguy) [reply demo.md:22]: Does this satisfy what your are saying?
```
  - Alice on Network1 can pay Bob on Network2 if the following conditions are met:
    - Alice has funds on Network1
    - Intermediary has funds on Network2
    - Intermediary has an exchange rate to convert between Network1 funds and Network2 funds
    - Intermediary can access both Network1 and Network2 to monitor Alice & Bob balances
```

Comment (geoknee) [reply demo.md:22]: Fair enough.

> Does this satisfy what you are saying?
> 
> ```
>   - Alice on Network1 can pay Bob on Network2 if the following conditions are met:
>     - Alice has sufficient funds on Network1
>     - Intermediary has sufficient funds on Network2
I would specify that these funds need to be in Bob's wallet. Not just "on Network2"


Otherwise, seems good. I don't know if you need to underline that the intermediary can access both networks. That is probably implied by having funds on each network.

Comment (bitwiseguy) [reply demo.md:22]: Good call, I'll remove the bullet about intermediary accessing both network. How about this:
```
  - Alice on Network1 can pay Bob on Network2 if the following conditions are met:
    - Alice has sufficient funds on Network1
    - Bob has designated enough of his Network2 funds for Intermediary
    - Intermediary has an exchange rate to convert between Network1 funds and Network2 funds
```

File: demo.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+- ERC-4337 compatible Smart Contract wallet
+- Unique feature: gasless off-chain txs with instant finality enabled by HTLC technology, as used in Bitcoin Lightning. Funds can be moved in and out of the off-chain payment system without any on-chain txs!
+- Setup required:
+  - on-chain: wallet owners must designate an Intermediary in their SC wallet
+  - off-chain: wallet owners and Intermediary must sign-off on some amount of wallet funds designated for off-chain txs
+  - run in-browser client(s) to interact with SC wallet and peers
+  - `Explain browser tab setup with Alice(left), Irene(middle), Bob(right)`
+- How it works:
+  - on-chain txs are 2/2 multi-sig (user + intermediary)
+  - `Show on-chain tx from Alice to Metamask account`
```
Comment (lalexgap) [inline demo.md:12]: It might be nice to do this on the receiving end, and have Bob show the transaction. We could have a flow like this:
- Alice and Bob start with a balance of [5,5]
- Alice pays Bob (crosschain!) an amount of 3
- We see all the balances update 
- Bob now has a balance of 8
- Bob sends an L1 transaction of 7 (which he couldn't afford before the payment) to some external address.

File: demo.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
+## Demo script:
+
+- ERC-4337 compatible Smart Contract wallet
+- Unique feature: gasless off-chain txs with instant finality enabled by HTLC technology, as used in Bitcoin Lightning. Funds can be moved in and out of the off-chain payment system without any on-chain txs!
```
Comment (lalexgap) [inline demo.md:4]: I think we should lean into the cross-chain aspect and also emphasize that. Not only do the payments happen completely off-chain, but the payments can be redeemed on different chains.

Comment (lalexgap) [inline demo.md:4]: We may want to be careful with the term `gasless`. While it's true the off-chain payments don't require gas, it could get confused with "gasless" ERC 4337 transactions, which is a whole different kettle of fish.

File: demo.md (hunk 5 above)
Comment (bitwiseguy) [reply demo.md:12]: Good call. That'd be a good way to show the cross-chain functionality

File: demo.md (hunk 6 above)
Comment (bitwiseguy) [reply demo.md:4]: But our txs are actually `gasless` whereas the ERC-4337 txs are more like "sponsored txs" where somebody else pays the gas 🙄. You are correct though, might be confusing to use that term. Originally I had "free off-chain txs". Do you like that better or can you think of any other term that might be better?

Comment (lalexgap) [reply demo.md:4]: Something like that makes sense to me! Maybe "free off-chain payments"? The term`Off-chain tx` seems like it might be slightly confusing, since `on-chain tx` let you do much more then pay someone.

Comment (bitwiseguy) [reply demo.md:4]: Agreed on the use of "off-chain payment" instead of "off-chain tx".

"free payment" seems contradictory but I'm good with it if you are 🤷‍♂️

File: demo.md (hunk 5 above)
Comment (geoknee) [reply demo.md:12]: Yes I like this!
